library/core/src/cell.rs

```diff
@@ -436,7 +435,10 @@ impl<T> Cell<T> {
     #[stable(feature = "move_cell", since = "1.17.0")]
     pub fn swap(&self, other: &Self) {
         fn is_nonoverlapping<T>(src: *const T, dst: *const T) -> bool {
-            intrinsics::is_nonoverlapping(src.cast(), dst.cast(), size_of::<T>(), 1)
+            let src_usize = src.addr();
+            let dst_usize = dst.addr();
+            let diff = src_usize.abs_diff(dst_usize);
+            diff >= size_of::<T>()
```

Comment:
Is there a specific reason why you are not calling the shared function here any more?

Reply:
`Cell::swap` documents that it _will_ panic if there is overlap, so even though it is not currently const, using the "only at runtime" function seems a bit fragile.

Reply:
Ah, fair. Please add a comment.